src/components/UserProfile/UserForm.tsx

```diff
@@ -1,77 +1,140 @@
 "use client";
 
-import type React from "react";
-import { useState } from "react";
+import { useEffect } from "react";
 import { useFormStatus } from "react-dom";
+import { useForm } from "react-hook-form";
 
 import { type Schema } from "@/amplify/data/resource";
 import { type UserFormProp } from "@/components/UserProfile/UserProfile";
+import { useUserDetails } from "@/components/contexts/UserDetailsContext";
+
+type UserFormData = {
+  id: string;
+  firstName: string;
+  lastName: string;
+  email: string;
+  institution: string;
+  willEatMeals: string;
+  allergies: string;
+};
+
+const getFormValues = (userDetails: any): UserFormData => ({
+  id: userDetails?.id || "",
+  firstName: userDetails?.firstName || "",
+  lastName: userDetails?.lastName || "",
+  email: userDetails?.email || "",
+  institution: userDetails?.institution || "",
+  willEatMeals: userDetails?.willEatMeals ? "true" : "false",
```

Comment:
Since we store `willEatMeals` in a boolean, there is no real need to actually store it as a string
```suggestion
    willEatMeals: Boolean(userDetails?.willEatMeals),
```